projects/storefrontlib/src/cms-components/product/product-variant-selector/color-selector/color-selector.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, ChangeDetectionStrategy, Input } from '@angular/core';
+import {
+  Product,
+  RoutingService,
+  BaseOption,
+  VariantQualifier,
+  VariantOptionQualifier,
+} from '@spartacus/core';
+
+@Component({
+  selector: 'cx-color-selector',
+  templateUrl: './color-selector.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VariantColorSelectorComponent {
+  constructor(private routingService: RoutingService) {}
+
+  @Input()
+  product: Product;
+
+  @Input()
+  variants: BaseOption;
+
+  changeSize(code: string): void {
```

Comment:
Is the method name `changeSize` correct for color selector? Maybe should we use `changeColor`.